ui/pages/confirmations/components/confirm/footer/footer.tsx

```diff
@@ -1,30 +1,55 @@
-import React from 'react';
+import { ethErrors, serializeError } from 'eth-rpc-errors';
+import React, { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 
 import {
   Button,
   ButtonSize,
   ButtonVariant,
 } from '../../../../../components/component-library';
 import { Footer as PageFooter } from '../../../../../components/multichain/pages/page';
+import { currentConfirmationSelector } from '../../../../../selectors';
+import {
+  rejectPendingApproval,
+  resolvePendingApproval,
+} from '../../../../../store/actions';
 import { useI18nContext } from '../../../../../hooks/useI18nContext';
 
 const Footer = () => {
   const t = useI18nContext();
+  const currentConfirmation = useSelector(currentConfirmationSelector);
+  const dispatch = useDispatch();
+
+  const onCancel = useCallback(() => {
+    if (!currentConfirmation) {
```

Comment:
@jpuri How can `currentConfirmation` be falsy here? If the state can indeed be that, shouldn't we disable the buttons instead?

Reply:
Hey @pedronfigueiredo : when the component is rendered initially `currentConfirmation` can be undefined for momentarily for just 1 render cycle (till the hook initialises current confirmation). The check above is to ensure that we do not get null error in that scenario.